modules/kubernetes-addons/prometheus/locals.tf

```diff
@@ -72,10 +47,36 @@ locals {
     operating_system = "linux",
   })]
 
+  amazon_prometheus_workspace_id           = var.amazon_prometheus_workspace_id != null && var.enable_amp_for_prometheus ? var.amazon_prometheus_workspace_id : null
```

Comment:
Isn't var.amazon_prometheus_workspace_id != null a sufficient check?

Reply:
yes that should do